distributed/spill.py

```diff
@@ -48,24 +48,25 @@ def __setitem__(self, key, value):
             )
             pass
         except OSError:
-            logger.warning(
-                "Spill file to disk failed",
-                exc_info=True,
-            )
+            logger.error("Spill to disk failed; keeping data in memory", exc_info=True)
             pass
         except PickleError as e:
-            logger.warning(
-                "Failed to pickle",
-                exc_info=True,
-            )
-            k, orig_e = e.args
-            if k == key:
-                del self[
-                    key
-                ]  # if the key that is bad is the one being added we delete it
+            key_e, orig_e = e.args
+            if key_e == key:
+                # The key we just inserted failed to serialize.
+                # This happens only when the key is individually larger than target.
+                # The exception will be caught by Worker and logged; the status of
+                # the task will be set to error.
```

Comment:
@cruzaderky according to this code we are not login the failed to pickle when key is individually larger then the target, is this on purpose? 
If so I need to modify the `test_spillbuffer_bad_key` as currently assumes we log for this case too. See https://github.com/ncclementi/distributed/blob/65815c13a0987f44983a73c29cdfbefddb1ec3eb/distributed/tests/test_spill.py#L216-L224